components/rhs_thread/rhs_thread.tsx

```diff
@@ -172,6 +172,19 @@ export default class RhsThread extends React.Component<Props, State> {
         }
     }
 
+    private userIsAtLastPost = () => {
+        const postReplies = (this.rhspostlistRef.current as HTMLElement)?.childNodes;
+        const rhsPostListRect = (this.rhspostlistRef.current as HTMLElement)?.getBoundingClientRect();
+        const offset = 24;
+
+        if (postReplies) {
+            const postRepliesArray = Array.from(postReplies);
+            return ((postRepliesArray[postRepliesArray.length - 1] as HTMLElement).offsetTop + offset) + rhsPostListRect.top === rhsPostListRect.bottom;
```

Comment:
I'm not sure I understand this logic here. It doesn't look like it's checking if the RHS is scrolled to the bottom since there would still be the comment box below `rhspostlistRef`. Could you explain what we're trying to check for here?

Reply:
Yeah it's definitely the offset for shorter posts, basically I check if the offset of the last post + its height minus the hidden part of the rhs post list container is equal to the visible distance.